packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -229,6 +228,118 @@ describe('HttpMocker', () => {
+                        surname: { type: 'string', default: 'Kent' },
+                        age: { type: ['number', 'null'] },
+                        email: { type: 'string' },
+                        deposit: { type: 'number' },
+                        paymentStatus: { type: 'string', enum: ['completed', 'outstanding'] },
+                        creditScore: {
+                          anyOf: [{ type: 'number', examples: [1958] }, { type: 'string' }],
+                        },
+                        paymentScore: {
+                          oneOf: [{ type: 'string' }, { type: 'number', examples: [1958] }],
+                        },
+                        walletScore: {
+                          allOf: [{ type: 'string' }, { default: 'hello' }],
+                        },
+                        pet: {
+                          type: 'object',
+                          properties: {
+                            name: { type: 'string', examples: ['Clark'] },
+                            middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                          },
+                        },
+                      },
+                      required: ['name', 'surname', 'age', 'email'],
+                    },
+                    encodings: [],
+                  },
+                ],
+              },
+            ],
+          };
+
+          const response = httpMocker.mock({
+            input: mockInput,
+            resource: resourceOperation,
+            config: { mock: { dynamic: false } },
+          });
+
+          describe('the property has an example key', () => {
+            it('should return the example key', () => expect(response.body).toHaveProperty('name', 'Clark'));
+            describe('and a default key', () => {
+              it('prefers the example', () => expect(response.body).toHaveProperty('middlename', 'J'));
+            });
+
+            describe('and the property containing the example is deeply nested', () => {
+              it('should return the example key', () => expect(response.body).toHaveProperty('pet.name', 'Clark'));
+              it('should still prefer the example', () => expect(response.body).toHaveProperty('pet.middlename', 'J'));
+            });
+          });
+
+          describe('the property has not an example, but a default key', () => {
+            it('should use such key', () => {
+              expect(response.body).toHaveProperty('surname', 'Kent');
+            });
+          });
+
+          describe('the property has nor default, nor example', () => {
+            describe('is nullable', () => {
+              it('should be set to null', () => expect(response.body).toHaveProperty('age', null));
+            });
+
+            describe('and is not nullable', () => {
+              it('should return the default string', () => expect(response.body).toHaveProperty('email', 'string'));
```

Comment:
Hmmm wait, I think that's incorrect.
The assumption was "dynamic = false"
`email: { type: 'string' }` does not have a static example...
I don't think returning `{ email: 'string' }` makes sense.
Email should not be included in the body at all.

Reply:
It really depends what we want to do with non required properties — this is a call for @philsturgeon

Reply:
I can live with it. For now! :P